### DISCUSSION TITLE
slugOptions registering only one language

### DISCUSSION THREAD

**Discussion by iforemann** [spatie/laravel-sluggable#225] [Q&A]
Hello all,
When I try to render a page it only registers one language even though I am using the translatable package.
I can describe the scenario like this:

Page Model;

```
<?php

namespace App\Shop\Pages;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Spatie\Sluggable\HasSlug;
use Spatie\Sluggable\SlugOptions;
use Spatie\Translatable\HasTranslations;

class Page extends Model
{
    use HasFactory, HasSlug, HasTranslations;

    public $translatable = ['title', 'slug', 'description', 'keywords', 'content'];

    protected $fillable = [
        'title',
        'slug',
        'description',
        'keywords',
        'image',
        'content',
        'is_home',
        'is_active',
        'footer',
        'url',
        'list_order'
    ];

    public function getSlugOptions() : SlugOptions
    {
        return SlugOptions::create()
            ->generateSlugsFrom('title')
            ->saveSlugsTo('slug');
    }
}
```

in PageController;

```
    public function store(PageRequest $request)
    {
       
        Page::create($request->all());

        return redirect()->route('admin.pages.index')->with('message', 'Create page successful!');
    }
```

and it given me this result;

![image](https://user-images.githubusercontent.com/14348709/151732071-26cf504a-dbb1-451f-bbfa-c5d8a0cf06e7.png)

Do you think this may cause?

**Comment by marvin-wtt** [comment]
You are using the wrong trait. Use `HasTranslateableSlug` in your Model instead of `HasSlug`

**Comment by cdt-team** [comment]
Hello, thanks to all the contributors for this package.

I have a error whit **HasTranslatableSlug**,

This's my model:

`<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Facades\Storage;
use Spatie\Sluggable\HasTranslatableSlug;
use Spatie\Sluggable\SlugOptions;
use Spatie\Translatable\HasTranslations;
use Spatie\EloquentSortable\Sortable;
use Spatie\EloquentSortable\SortableTrait;

class SoftwareFor extends Model
{
    use HasFactory, HasTranslations, HasTranslatableSlug, SortableTrait;

    //Sortable field
    public $sortable = [
        'order_column_name' => 'order',
        'sort_when_creating' => false,
    ];

    public $translatable = ['pre_title', 'intro', 'suitable_for', 'title', 'body', 'slug'];


    protected $table = 'software_for_who';

    /**
     * Get the options for generating the slug.
     */
    public function getSlugOptions() : SlugOptions
    {

        return SlugOptions::createWithLocales(['it', 'en'])
            ->generateSlugsFrom(function($model, $locale) {
                return "{$locale} {$model->id}";
            })
            ->saveSlugsTo('slug');
    }
}
`

This is the error:

`[2022-02-10 15:43:58] local.ERROR: SQLSTATE[22032]: <<Unknown error>>: 3141 Invalid JSON text in argument 1 to function json_extract: "The document is empty." at position 0. (SQL: select exists(select * from `software_for_who` where json_unquote(json_extract(`slug`, '$."it"')) = it-7 and `id` != 7) as `exists`) {"userId":1,"exception":"[object] (Illuminate\\Database\\QueryException(code: 22032): SQLSTATE[22032]: <<Unknown error>>: 3141 Invalid JSON text in argument 1 to function json_extract: \"The document is empty.\" at position 0. (SQL: select exists(select * from `software_for_who` where json_unquote(json_extract(`slug`, '$.\"it\"')) = it-7 and `id` != 7) as `exists`) at /usr/share/nginx/html/site/vendor/laravel/framework/src/Illuminate/Database/Connection.php:712)`

Someone who can help?

**Reply by appfolder** [reply]
+1
